contracts/staking/GoodBatStaking.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.7.0;
+import "./SimpleStaking.sol";
+import "../Interfaces.sol";
+
+/**
+ * @title Staking contract that donates earned interest to the DAO
+ * allowing stakers to deposit BAT
+ * or withdraw their stake in BAT
+ * the contracts buy cBat and can transfer the daily interest to the  DAO
+ */
+contract GoodBatStaking is SimpleStaking {
```

Comment:
i dont understand why we have this contract

Reply:
To have different contract beside DAI cDAI

Reply:
we have GoodCompoundStaking
that's a generic compound staking for ALL cTokens, you simply initialize GoodCompoundStaking with the BAT token details